egov/egov-egi/src/main/java/org/egov/infra/admin/master/service/UserService.java

```diff
@@ -180,14 +190,72 @@ else if (user.getType().equals(UserType.EMPLOYEE))
         createUserRequest.setRequestInfo(microserviceUtils.createRequestInfo());
 
         final RestTemplate restTemplate = new RestTemplate();
+        UserDetailResponse udr = null;
         try {
-            restTemplate.postForObject(createUserServiceUrl, createUserRequest, UserDetailResponse.class);
+            udr = restTemplate.postForObject(createUserServiceUrl, createUserRequest, UserDetailResponse.class);
         } catch (final Exception e) {
             final String errMsg = "Exception while creating User in microservice ";
             LOGGER.error(errMsg, e);
             throw new ApplicationRuntimeException(errMsg, e);
         }
-        return userRepository.save(user);
+        final Long userId = udr.getUser().get(0).getId();
```

Comment:
Is there any issue with  userRepository.save(user); ? we can't get rid of Hibernate at any cost, keeping both doesn't suits well for erp.